### PR TITLE
Add deep link and message notification events (close #143, #142)

### LINKED ISSUE #143: Add deep link event and context
This was introduced in v3 of mobile trackers.

### PR DESCRIPTION
**Pull request by matus-tomlein**
This PR adds new event types introduced in the v3 of native trackers:

* deep link event and context – can be used by users to track when deep links are received. The context is attached to subsequent screen views tracked after the deep link received event,
* message notification event to track received notifications by users.

### CODE REVIEW

File: tests/__tests__/events.test.ts [hunk 1]
```diff
@@ -394,6 +394,34 @@ describe('test validateConsentWithdrawn', () => {
...
+  });
+
+  test('test invalid deepLinkReceived - invalid url', async () => {
+    const testEv = {
+      url: true
+    } as any;
+    await expect(e.validateDeepLinkReceived(testEv)).rejects.toThrow(logMessages.deepLinkReq);
+  });
+
+  test('test valid consentGranted', async () => {
```
Comment (adatzer) [inline tests/__tests__/events.test.ts]: nit - `'test valid deepLinkReceived'`

File: src/types.ts [hunk 2]
```diff
@@ -532,6 +537,113 @@ export type EcommerceTransactionProps = {
...
+   */
+   titleLocArgs?: string[];
+  /**
+   * The key to the title string in the app's string resources to use to localize the title text to the user's current localization.
+   */
+   titleLocKey?: string;
+  /**
+   * The trigger that raised the notification message. Must be one of: push, location, calendar, timeInterval, other
+   */
+   trigger: string;
```
Comment (adatzer) [inline src/types.ts]: Would it make sense to make a `Trigger` type like we have for other enums?

File: src/events.ts [hunk 3]
```diff
@@ -202,6 +204,51 @@ function validateConsentWithdrawn(argmap: ConsentWithdrawnProps): Promise<boolea
...
+  // validate type
+  if (!isObject(argmap)) {
+    return Promise.reject(new Error(logMessages.evType));
+  }
+  // validate required props
+  if (
+    typeof argmap.title !== 'string'
+    || typeof argmap.body !== 'string'
+    || typeof argmap.trigger !== 'string'
+    || !['push', 'location', 'calendar', 'timeInterval'].includes(argmap.trigger)
```
Comment (adatzer) [inline src/events.ts]: `'other'` is also one of the enum values for trigger

File: ios/RNSnowplowTracker.m [hunk 4]
```diff
@@ -433,6 +435,148 @@ @implementation RNSnowplowTracker
...
+        }
+        NSString *bodyLocKey = [argmap sp_stringForKey:@"bodyLocKey" defaultValue:nil];
+        if (bodyLocKey) {
+            event.bodyLocKey = bodyLocKey;
+        }
+        NSString *category = [argmap sp_stringForKey:@"category" defaultValue:nil];
+        if (category) {
+            event.category = category;
+        }
+        NSNumber *contentAvailable = [argmap sp_numberForKey:@"contentAvailable" defaultValue:nil];
```
Comment (adatzer) [inline ios/RNSnowplowTracker.m:526]: not sure how this plays out with booleans, how about `sp_boolForKey`?

**Review by adatzer — COMMENTED** [review]
Looks great @matus-tomlein !

File: tests/__tests__/events.test.ts (hunk 1 above)
Comment (matus-tomlein) [reply tests/__tests__/events.test.ts]: Thanks, good catch!

File: src/types.ts (hunk 2 above)
Comment (matus-tomlein) [reply src/types.ts]: I was considering enums but didn't notice how they are implemented in types.ts, that's much better! Changed it, thanks.

File: src/events.ts (hunk 3 above)
Comment (matus-tomlein) [reply src/events.ts]: Lifesaver!

File: ios/RNSnowplowTracker.m (hunk 4 above)
Comment (matus-tomlein) [reply ios/RNSnowplowTracker.m:526]: NSNumber also supports booleans. We could also use `sp_boolForKey` but the `event.contentAvailable` property takes NSNumber so we would have to transform to NSNumber anyway.

**Review by adatzer — APPROVED** [review]
LGTM!!

**Pull request merged**